src/OpenApi/src/Extensions/OpenApiServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,72 @@
+    public static IServiceCollection AddOpenApi(this IServiceCollection services, string documentName)
+    {
+        ArgumentNullException.ThrowIfNull(services);
+
+        return services.AddOpenApi(documentName, _ => { });
+    }
+
+    /// <summary>
+    /// Adds OpenAPI services related to the given document name to the specified <see cref="IServiceCollection"/> with the specified options.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register services onto.</param>
+    /// <param name="documentName">The name of the OpenAPI document associated with registered services.</param>
+    /// <param name="configureOptions">A delegate used to configure the target <see cref="OpenApiOptions"/>.</param>
+    public static IServiceCollection AddOpenApi(this IServiceCollection services, string documentName, Action<OpenApiOptions> configureOptions)
+    {
+        ArgumentNullException.ThrowIfNull(services);
+        ArgumentNullException.ThrowIfNull(configureOptions);
+
+        services.AddOpenApiCore(documentName);
+        services.Configure<OpenApiOptions>(documentName, options =>
+        {
+            options.DocumentName = documentName;
+            configureOptions(options);
+        });
+        return services;
+    }
+
+    /// <summary>
+    /// Adds OpenAPI services related to the default document to the specified <see cref="IServiceCollection"/> with the specified options.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register services onto.</param>
+    /// <param name="configureOptions">A delegate used to configure the target <see cref="OpenApiOptions"/>.</param>
+    public static IServiceCollection AddOpenApi(this IServiceCollection services, Action<OpenApiOptions> configureOptions)
+            => services.AddOpenApi(OpenApiConstants.DefaultDocumentName, configureOptions);
+
+    /// <summary>
+    /// Adds OpenAPI services related to the default document to the specified <see cref="IServiceCollection"/>.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register services onto.</param>
+    public static IServiceCollection AddOpenApi(this IServiceCollection services)
+        => services.AddOpenApi(OpenApiConstants.DefaultDocumentName);
+
+    private static IServiceCollection AddOpenApiCore(this IServiceCollection services, string documentName)
+    {
+        services.AddEndpointsApiExplorer();
+        services.AddKeyedSingleton<OpenApiComponentService>(documentName);
+        services.AddKeyedSingleton<OpenApiDocumentService>(documentName);
+        // Required for build-time generation
+        services.AddSingleton<IDocumentProvider, OpenApiDocumentProvider>();
+        // Required to resolve document names for build-time generation
+        services.AddSingleton(new NamedService<OpenApiDocumentService>(documentName));
```

Comment:
Are any of these services something a user would override? If so we should probably use `TryAddSingleton`

Reply:
Not really. All of these types are internal and will be for the foreseeable future.